afl-fuzz.c

```diff
@@ -8071,6 +8094,11 @@ int main(int argc, char** argv) {
 
   if (queue_cur) show_stats();
 
+  /* Now that we've killed the forkserver, we wait for it to be able to get rusage stats. */
```

Comment:
There are 4 different ways to have stop_soon break out of the loop: handle_stop_sig when user sends SIGINT for example; via AFL_EXIT_WHEN_DONE when there's been no progress in the fuzzing for several cycles; via AFL_BENCH_UNTIL_CRASH after the first crash; and AFL_BENCH_JUST_ONE after one cycle.

The forkserver is only killed via handle_stop_sig where stop_soon will be set to 1, so in the other cases we need to kill the forkserver and its childs before the waitpid.